{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "![Harvard Seal](./img/HMS_Seal.png)\n",
    "\n",
    "2nd Day\n",
    "---\n",
    "\n",
    "> Enes Kemal Ergin\n",
    "\n",
    "> 11/04/2015\n",
    "\n",
    "Outline of the Day\n",
    "\n",
    "- Converted files to bedgraph format\n",
    "- Got only chromosome 1 from Human Genome\n",
    "- Got only chromosome 1 from HeLA cells\n",
    "- Learned how to use command line cooler way\n",
    "- Used genomedata to create a dataset from HG, and HeLA Cells\n",
    "- Small session with Dr.Umut\n",
    " "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Converting to BedGraph\n",
    "---\n",
    "At the very beginning of the day we starting working on the files that I downloaded yesterday. When we try to run those data with genomedata to combine them, we couldn't be able to do so, since the format of the HeLA cells are not compatible in genomedata. \n",
    "\n",
    "Then I found a tool to convert ```BigWig``` files to ```bedgraph (bg)``` file. Tool was just an executable file to run with 2 different files, here is how I did it.\n",
    "\n",
    "```sh\n",
    "bigwigToBedGraph file1.bigwig file2.bg\n",
    "```\n",
    "\n",
    "I run this command for 2 heLa cells. After sometime(since the files were very big), I got 2 bedgraph files. Using command line is very good to see what huge files have inside:\n",
    "\n",
    "```sh\n",
    "less file2.bg\n",
    "```\n",
    "\n",
    "Output:\n",
    "\n",
    "```\n",
    "chr1    9950    9975    0.16\n",
    "chr1    9975    10025   1\n",
    "chr1    10025   10050   1.72\n",
    "chr1    10050   10150   2\n",
    "chr1    10150   10175   1.84\n",
    "chr1    10175   10225   1\n",
    "chr1    10225   10250   0.28\n",
    "chr1    10350   10375   0.8\n",
    "chr1    10375   10550   1\n",
    "chr1    10550   10575   0.2\n",
    "chr1    11075   11100   0.64\n",
    "chr1    11100   11275   1\n",
    "chr1    11275   11300   0.36\n",
    "chr1    11375   11400   0.88\n",
    "chr1    11400   11425   2.16\n",
    "chr1    11425   11500   3\n",
    "chr1    11500   11525   3.12\n",
    "chr1    11525   11550   4\n",
    "chr1    11550   11575   4.04\n",
    "chr1    11575   11600   4.68\n",
    "...\n",
    "```\n",
    "\n",
    "--- "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Slicing the files\n",
    "---\n",
    "\n",
    "Since files that we are working on is very huge it consumes a lot of time. That's why I slice the data in between chromosome 1 and 2 and store it in another file. \n",
    "\n",
    "Here is my notes for slicing HumanGenome and HeLA cells:\n",
    "\n",
    "```sh\n",
    "# How to modify the HumanGenome38 form that task asks for\n",
    "\n",
    "\n",
    "# Create a new file with the lines between chromosome 1 and chromosome 2\n",
    "sed -n \"/.*chromosome 1.*/,/.*chromosome 2.*/p\" HumanGenome38.fa > hg38_chr1.fa\n",
    "\n",
    "# Need to Extract the last line since it contains chromosome 2\n",
    "sed -i '$ d' hg38_chr1.fa\n",
    "\n",
    "\n",
    "# Also for the task we need to have only > chr 1 for the description\n",
    "# Make a temporary file to append them all\n",
    "echo \">chr1\" > chr1.fa\n",
    "\n",
    "# get the length of the lines in seq\n",
    "wc -l hg38_chr1.fa\n",
    "\n",
    "# 17155298 in our case so  we will append the lines except the first line \n",
    "tail -17155297 hg38_chr1.fa >> chr1.fa\n",
    "\n",
    "# At the end we get the fasta file that we want.\n",
    "\n",
    "\n",
    "## How to get only chr1 from HeLa cells using Terminal\n",
    "\n",
    "# Takes only chromosome 1 from whole data\n",
    "grep -P 'chr1\\t' test27.bg > K27ac_chr1.bg\n",
    "\n",
    "# Takes only chromosome 1 from whole data\n",
    "grep -P 'chr1\\t' test36.bg > K36me3_chr1.bg\n",
    "\n",
    "# Now we have two HeLa cells which has only chr1 information available\n",
    "```"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "Writing all those commands gave me an opportunity to learn grep, sed, awk to make terminal is cooler than ever\n",
    "\n",
    "Then run genomedata with data that we just sliced:\n",
    "\n",
    "```sh\n",
    "## Creating a sample dataset using data that we modified:\n",
    "\n",
    "# Need genomedata-load to load the datasets\n",
    "\n",
    "# Command: genomedata-load \n",
    "# Sequence: chr1.fa (Human Genome)\n",
    "# Tracks: \n",
    "        # K27ac: K27ac_chr1.bg \n",
    "        # K36me: K36me3_chr1.bg\n",
    "# Dataset Name: genomedataset.test\n",
    " \n",
    "genomedata-load -s chr1.fa -t K27ac=K27ac_chr1.bg -t K36me3=K36me3_chr1.bg  genomedataset.test\n",
    "```\n",
    "\n",
    "It took quite sometime to finish executing. At the end we will get ```genomedataset.test``` folder.\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "After finishing all those convertion and slicing, we went to the next step which is working on the project folder that we got day before. Dr. Umut explained the scripts and the overall picture of the project using computational approaches. I am not going to get into more detail since it is confidential for now...\n",
    "\n",
    "---\n",
    "\n",
    "See you tomorrow..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
